search-handlers/src/test/java/no/unit/nva/search/SearchResourceAuthHandlerTest.java

```diff
@@ -86,20 +90,99 @@ void shouldReturnOkWhenUserIsEditor() throws IOException {
     prepareRestHighLevelClientOkResponse();
 
     var input =
-        getInputStreamWithAccessRight(randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL);
+        getInputStreamWithAccessRight(
+            randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL, V_LEGACY);
+    handler.handleRequest(input, outputStream, contextMock);
+
+    var gatewayResponse = FakeGatewayResponse.of(outputStream);
+
+    assertNotNull(gatewayResponse.headers());
+    assertEquals(HTTP_OK, gatewayResponse.statusCode());
+  }
+
+  @Test
+  void shouldDefaultToLegacyModelWhenNoVersionSpecified() throws IOException {
+    prepareRestHighLevelClientOkResponse();
+
+    var input =
+        getInputStreamWithAccessRight(
+            randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL, null);
+    handler.handleRequest(input, outputStream, contextMock);
+
+    var gatewayResponse = FakeGatewayResponse.of(outputStream);
+
+    assertNotNull(gatewayResponse.headers());
+    assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    assertNotNull(gatewayResponse.body());
+    assertThat(gatewayResponse.body().hits().size(), is(equalTo(2)));
+
+    var firstHitJson = gatewayResponse.body().hits().getFirst();
+    assertLegacyModel(firstHitJson);
```

Comment:
This is asserting many things, are they all relevant?

- If they are, maybe split it into multiple tests?
- Why assert exactly 2 responses?
- Is it necessary to assert that the response is OK with a body etc, when you then get the response body and verify the content is OK?